src/dir.rs

```diff
@@ -1,36 +1,50 @@
-use crate::types::{Epoch, Rank};
 use serde::{Deserialize, Serialize};
-use std::path::Path;
+use std::path::PathBuf;
+
+pub use f64 as Rank;
+pub use i64 as Epoch; // use a signed integer so subtraction can be performed on it
```

Comment:
I think this comment is unnecessary. Just a suggestion; feel free to disregard if you think otherwise.

```suggestion
pub use i64 as Epoch;
```

Reply:
I left that in there because it can be tempting to change it to a `u64`, since that is what `Duration::as_secs` returns. However, this could potentially let errors go undetected, so it's better to be explicit here.